doc/guides/decoding/decoder_plugins.rst

```diff
@@ -0,0 +1,129 @@
+.. _guide_decoder_plugins:
+
+==========================
+Pixel Data Decoder Plugins
+==========================
+
+*Pixel Data* decoding in *pydicom* uses an :class:`~pydicom.pixels.decoders.base.Decoder`
+instance for the specific *Transfer Syntax* as a manager for plugins that
+perform the decoding work. This guide covers the requirements for those plugins
+and how to add them to *pydicom*.
+
```

Comment:
Just for people stumbling onto this doc page, it might be nice to point out that this is really for advanced users needing something other than the available decoders (link to them).